HACKING

```diff
@@ -633,6 +633,41 @@ When XCL_NOT is set, the bit map, if present, contains bits for characters that
 are allowed (exactly as for OP_NCLASS), but the list of items that follow it
 specifies characters and properties that are not allowed.
 
+The meaning of the bitmap indicated by XCL_MAP is that, if one is present, then
+it fully describes which code points < 256 match the class (without needing to
+invert the check according to XCL_NOT); the other items in the OP_XCLASS need
+not be consulted. However, if a bitmap is not present, then code points < 256
+may still match, so the other items in the OP_XCLASS must be consulted.
```

Comment:
Would someone mind checking I've understood correctly the behaviour of the bitmap for OP_XCLASS?

I made a bug in an earlier version of the PR, where I had assumed that, if an XCLASS had no bitmap, then it wouldn't match any ASCII characters. I now realise that is wrong.

If we're willing to spend 32 bytes though, we could speed up matching of things like `\pL` if we _did_ include the bitmap for all cases of XCLASS which match in the ASCII range.

Reply:
Bitmap is generated, if cranges has a range, which start is less than 256. The \p or \P properties might match to characters < 256 even if there is no bitmap.

Reply:
Great! That matches my understanding now.